### DISCUSSION TITLE
question about version 38794

### DISCUSSION THREAD

**Discussion by Roberto8303** [FreeCAD/FreeCAD#16773] [Code contribution discussions]
I wanted to ask in the last revision 38794 what was fixed? because I don't see any changes

**Comment by FlachyJoe** [comment]
Between 38794 and what?
https://github.com/FreeCAD/FreeCAD/commits/main/

**Comment by Roberto8303** [comment]
vers 38643

**Reply by luzpaz** [reply]
`git log Rev1...Rev2` should work but it doesn't work on my git setup

@FlachyJoe can you try this and see what output you get? 

`git log --oneline 38794..38643`

**Reply by oursland** [reply]
The numbers aren't SHAs, but rather the commit number as computed by `git rev-list --count 7d8e53aaab17961d85c5009de34f69f2af084e8b..HEAD` + 14555. [source](https://github.com/FreeCAD/FreeCAD/blob/main/src/Tools/SubWCRev.py#L294)

The result is that we can generate a monotonically increasing numeric value for any given commit, however it is not guaranteed to be unique and isn't easily back-mapped to the commit associated with that number.  To compute the diff though this number is non-trivial.

**Reply by luzpaz** [reply]
@oursland so if I understand you correctly using the numeric value of a commit isn't reliable?

**Reply by FlachyJoe** [reply]
This gives the commit hash from the rev number
```bash
#!/bin/bash
currentNumber=$[$(git rev-list --count 7d8e53aaab17961d85c5009de34f69f2af084e8b..HEAD) + 14555]
echo $(git rev-list 7d8e53aaab17961d85c5009de34f69f2af084e8b..HEAD | sed -n "$[$currentNumber - $1]p")
```
